tests/common/test_cgroups.py

```diff
@@ -34,262 +31,131 @@ def consume_cpu_time():
     return waste
 
 
-def make_self_cgroups():
-    """
-    Build a CGroups object for the cgroup to which this process already belongs
+def i_am_root():
+    return os.geteuid() == 0
 
-    :return: CGroups containing this process
-    :rtype: CGroupConfigurator
-    """
 
-    def path_maker(hierarchy, __):
-        suffix = CGroupConfigurator._get_current_process_cgroup_path(CGroupConfigurator.get_hierarchy_id('cpu'))
-        return os.path.join(BASE_CGROUPS, hierarchy, suffix)
+class TestCGroup(AgentTestCase):
+    def tearDown(self):
+        AgentTestCase.tearDown(self)
 
-    return CGroupConfigurator("inplace", path_maker)
+        with open(os.path.join(data_dir, "cgroups", "cpu_mount", "tasks"), mode="wb") as tasks:
+            tasks.truncate(0)
+        with open(os.path.join(data_dir, "cgroups", "memory_mount", "tasks"), mode="wb") as tasks:
+            tasks.truncate(0)
 
+    def test_correct_creation(self):
+        test_cgroup = CGroup.create("dummy_path", "cpu", "test_extension")
+        self.assertIsInstance(test_cgroup, CpuCgroup)
+        self.assertEqual(test_cgroup.controller, "cpu")
+        self.assertEqual(test_cgroup.path, "dummy_path")
+        self.assertEqual(test_cgroup.name, "test_extension")
 
-def make_root_cgroups():
-    """
-    Build a CGroups object for the topmost cgroup
+        test_cgroup = CGroup.create("dummy_path", "memory", "test_extension")
+        self.assertIsInstance(test_cgroup, MemoryCgroup)
+        self.assertEqual(test_cgroup.controller, "memory")
+        self.assertEqual(test_cgroup.path, "dummy_path")
+        self.assertEqual(test_cgroup.name, "test_extension")
 
-    :return: CGroups for most-encompassing cgroup
-    :rtype: CGroupConfigurator
-    """
+    def test_is_active(self):
+        test_cgroup = CGroup.create(os.path.join(data_dir, "cgroups", "cpu_mount"), "cpu", "test_extension")
+        self.assertEqual(False, test_cgroup.is_active())
 
-    def path_maker(hierarchy, _):
-        return os.path.join(BASE_CGROUPS, hierarchy)
+        with open(os.path.join(data_dir, "cgroups", "cpu_mount", "tasks"), mode="wb") as tasks:
+            tasks.write(str(1000).encode())
 
-    return CGroupConfigurator("root", path_maker)
+        self.assertEqual(True, test_cgroup.is_active())
 
+        test_cgroup = CGroup.create(os.path.join(data_dir, "cgroups", "memory_mount"), "memory", "test_extension")
+        self.assertEqual(False, test_cgroup.is_active())
 
-def i_am_root():
-    return os.geteuid() == 0
+        with open(os.path.join(data_dir, "cgroups", "memory_mount", "tasks"), mode="wb") as tasks:
+            tasks.write(str(1000).encode())
+
+        self.assertEqual(True, test_cgroup.is_active())
+
+    def test_is_active_incorrect_file(self):
+        test_cgroup = CGroup.create(os.path.join(data_dir, "cgroups", "not_cpu_mount"), "cpu", "test_extension")
+        self.assertEqual(False, test_cgroup.is_active())
```

Comment:
I fail to see what this test is for, since the assert would always be `False` because there are no tasks in the `tasks` file. Maybe a better assert would be that nothing was returned?